packages/authentication/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/authentication/FirebaseAuthenticationPlugin.java

```diff
@@ -984,6 +985,23 @@ public void handleAuthStateChange() {
         notifyListeners(AUTH_STATE_CHANGE_EVENT, result, true);
     }
 
+    public void handleIdTokenChange() {
+        NonEmptyResultCallback callback = new NonEmptyResultCallback() {
+            @Override
+            public void success(Result result) {
+                notifyListeners(ID_TOKEN_CHANGE_EVENT, result.toJSObject(), true);
+            }
+
+            @Override
+            public void error(Exception exception) {
+                JSObject changeResult = new JSObject();
+                changeResult.put("token", "");
+                notifyListeners(ID_TOKEN_CHANGE_EVENT, changeResult, true);
```

Comment:
```suggestion
                Logger.error(TAG, exception.getMessage(), exception);
```

Let's log errors first and wait for feedback. Maybe it makes sense to add an `idTokenChangeError` event listener later.